eth/backend.go

```diff
@@ -259,7 +274,23 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 
 	peers := newPeerSet()
 	bcOps = append(bcOps, core.EnableBlockValidator(chainConfig, eth.engine, config.TriesVerifyMode, peers))
-	eth.blockchain, err = core.NewBlockChain(chainDb, cacheConfig, config.Genesis, &overrides, eth.engine, vmConfig, eth.shouldPreserve, &config.TransactionHistory, bcOps...)
+	txLookupLimit := &config.TransactionHistory
+	// if enable HistorySegment, just skip txLookupLimit params,
+	// may cause regenerate tx index, but it will also generate new block index
+	if config.HistorySegmentEnabled {
+		txLookupLimit = nil
```

Comment:
why disable transaction index generation when this BEP is enabled?

Reply:
It is a very simple way to handle the tx index, it prevents the indexer from regenerating ancient tx index, and it may cause error when ancient data has been removed.